tests/thread_local_tests/thread_local_test/tests.cpp

```diff
@@ -0,0 +1,76 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#include <gtest/gtest.h>
+#include <thread>
+#include <future>
+#include <windows.h>
+#include <psapi.h>
+#include <tlhelp32.h>
+
+class ThreadLocalTest : public ::testing::Test, public ::testing::WithParamInterface<std::string>
+{
+public:
+    void SetUp()
+    {
+        target_device = GetParam();
+    }
+
+    static std::string getTestCaseName(testing::TestParamInfo<std::string> obj)
+    {
+        const auto layerName = obj.param;
+        std::ostringstream result;
+
+        result << layerName;
+        return result.str();
+    }
+
+public:
+    std::string target_device = "";
+};
+
+TEST(LoadLibraryTest, load_free_library)
+{
+    std::promise<void> free_promise;
+    std::future<void> free_future = free_promise.get_future();
+    std::promise<void> thread_exit_promise;
+    std::future<void> thread_exit_future = thread_exit_promise.get_future();
+    HMODULE shared_object = LoadLibraryA("openvino.dll");
+
+    std::thread sub_thread = std::thread([&]
+                                         {
+        free_promise.set_value();
+        thread_exit_future.get(); });
+    free_future.get();
+    FreeLibrary(shared_object);
+
+    thread_exit_promise.set_value();
+    if (sub_thread.joinable())
+    {
+        sub_thread.join();
+    }
+}
+
+typedef void (*TEST_FUNC)(std::string);
```

Comment:
```suggestion
typedef void (*TestFunc)(std::string);
```
Is more common for type in OV.
why pass string by value? Use const reference?